www/js/app.js

```diff
@@ -926,33 +948,35 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
 
     /**
      * Handle key input in the prefix input zone
-     * @param {Event} evt
+     * @param {Event} evt The event data to handle
      */
     function onKeyUpPrefix(evt) {
         // Use a timeout, so that very quick typing does not cause a lot of overhead
         // It is also necessary for the words suggestions to work inside Firefox OS
-        if(window.timeoutKeyUpPrefix) {
+        if (window.timeoutKeyUpPrefix) {
             window.clearTimeout(window.timeoutKeyUpPrefix);
         }
-        window.timeoutKeyUpPrefix = window.setTimeout(function() {
+        window.timeoutKeyUpPrefix = window.setTimeout(function () {
             var prefix = $("#prefix").val();
-            if (prefix && prefix.length>0) {
+            if (prefix && prefix.length > 0 && prefix !== global.search.prefix) {
                 $('#searchArticles').click();
             }
-        }
-        ,500);
+        }, 500);
     }
 
-
     /**
      * Search the index for DirEntries with title that start with the given prefix (implemented
      * with a binary search inside the index file)
-     * @param {String} prefix
+     * @param {String} prefix The string that must appear at the start of any title searched for
      */
     function searchDirEntriesFromPrefix(prefix) {
         if (selectedArchive !== null && selectedArchive.isReady()) {
+            // Cancel any previous search that may still be running before creating new search
+            global.search.state = 'cancelled';
+            // Store the new search term in the state.search global object and initialize
+            global.search = {'prefix': prefix, 'state': 'init', 'type': ''};
```

Comment:
@mossroy Can you please advise me on the above four lines? This is the bit that's worrying me (and the reason why I originally kept a list of searches). Here, we cancel a running search, and the cancelled state is reflected immediately in zimArchive.js. However, in the very next line, we store a new object in `global.search`. I don't have clear in my head what happens to the previously cancelled object that had been passed to `zimArchive.js`. Is that version somehow delinked from `global.search` but still exists in a delinked state?

Reply:
OK, @mossroy , yes that was a very rookie bit of code I'm embarrassed about now I see it. It arose from my removing the list of searches and not checking through the logic.... doh. Thanks for code suggestions, I see how that works. Will hopefully finally get this right...